SPEC.md

```diff
@@ -9883,15 +9884,20 @@ Test config:
 
 ```
 Array[P] keys(Map[P, Y])
+Array[String] keys(Struct|Object)
 ```
 
-Creates an `Array` of the keys from the input `Map`, in the same order as the elements in the map.
+Given a key-value type collection (`Map`, `Struct`, or `Object`), returns an `Array` of the keys from the input collection, in the same order as the elements in the collection.
```

Comment:
If the key of the input is optional then does the returned Array contain a concrete String?

For example:

```
Struct Name {
    last: String,
    first: String,
    middle: String?
}

...
Name name = {
    first: "John",
    last: "Doe"
}

Array[String] the_keys = keys(name).
```
Does `the_keys` contain the `String middle` or not?

Reply:
Yes, for a struct all members are always present, and optional members default to `None` if a value is not specified.